scripts/ci/python_native_extension_peer_gate.py

```
@@ -0,0 +1,583 @@
+
+    if not log_text or any(marker in log_text for marker in FORBIDDEN_LOG_MARKERS):
+        return False
+    if re.search(r"^FAILED\s+", log_text, re.MULTILINE):
+        return False
+
+    missing_modules = MISSING_MODULE_RE.findall(log_text)
+    partial_native_imports = PARTIAL_NATIVE_IMPORT_RE.findall(log_text)
+    collection_errors = COLLECTION_ERROR_RE.findall(log_text)
+    interruptions = INTERRUPTED_RE.findall(log_text)
+    if (
+        not missing_modules and not partial_native_imports
```

Ruff (0.16.2)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)